### PR TITLE
v1.4.1

### PR DESCRIPTION
**Pull request by JoaoAndreSa**
# Change Log
All notable changes introduced by this new version are documented here.
 
### Added

- Added a new global timeout for all node communication that can be configured by setting the environment variable `CONN_TIMEOUT` 

### Changed
  
- Updated all dependencies (particularly important is the new version of _Onet 3.2.0_ that includes better error tracking)
- Switched from `errors.new` to `fmt.Errorf` (the latter keeps the stack trace)
- Removed the redundant `isLeaf()` condition for the `SendToChildren` call
- Renamed some `tmp` variables for easy code readability
- Conditions no longer use `== false` or `== true` 
- Renamed main exec package to `main`
- Reorganized the app folder files: `/app/ ` is changed to `/cmd/[application_name]/`
 
### Fixed
 
- Fixed the error generated by the `tn.Config()` function by moving it out of the `NewProtocol(tn *onet.TreeNodeInstance, conf *onet.GenericConfig)` and adding it to right before this function call. _"...this way, only the root will actually set the config for the whole tree, ensuring that every node gets it before starting._"

### CODE REVIEW

**Comment by JoaoAndreSa** [comment]
Quick question do you know how to remove this ci/dockercloud check? I tried changing the branch protection rule, but it's still here...

**Comment by mickmis** [comment]
> Quick question do you know how to remove this ci/dockercloud check? I tried changing the branch protection rule, but it's still here...

It was the Docker Cloud webhook, I removed it so it should be gone now.

File: lib/constants.go [hunk 1]
```diff
@@ -1,30 +1,44 @@
...
+	"go.dedis.ch/onet/v3/log"
+	"os"
 	"sync"
+	"time"
 
 	"go.dedis.ch/onet/v3/simul/monitor"
 )
 
+func init() {
+	timeout, err := time.ParseDuration(os.Getenv("MEDCO_TIMEOUT"))
```
Comment (mickmis) [inline lib/constants.go]: Is it wise here to mention MedCo? Shouldn't Unlynx be an independent piece of software?

Comment (JoaoAndreSa) [reply lib/constants.go]: I can rename this environnent variable :) Any suggestions? Maybe COM_TIMEOUT (for communication timeout)?

Comment (mickmis) [reply lib/constants.go]: CONN_TIMEOUT ? :)

Comment (JoaoAndreSa) [reply lib/constants.go]: done yours is a better name!

**Pull request merged**